{{cookiecutter.project_slug}}/frontend/app/lib/utilities/posts.ts

```diff
@@ -0,0 +1,133 @@
+import fs from "fs"
+import path from "path"
+import matter from "gray-matter"
+import { remark } from "remark"
+import html from "remark-html"
+import remarkGfm from "remark-gfm"
+import remarkToc from "remark-toc"
+
+const postsDirectory = path.join(process.cwd(), "app/content/blog")
+
+// -------------------------------------------------
+// GET THE DATA OF ALL POSTS IN SORTED ORDER BY DATE
+/*
+  Returns an array that looks like this:
+  [
+    {
+      id: 'ssg-ssr',
+      title: 'When to Use Static Generation v.s. Server-side Rendering',
+      date: '2020-01-01'
+    },
+    {
+      id: 'pre-rendering',
+      title: 'Two Forms of Pre-rendering',
+      date: '2020-01-02'
+    }
+  ]
+*/
+
```

Comment:
Not sure if this is another docstring formatting for JS, but this is the one I commonly see. 
```
/**
 * [bar description]
 * @param  {[type]} foo [description]
 * @return {[type]}     [description]
 */
```
```suggestion
/**
 * Get the data of all posts in sorted order by date
 * @return {List[Object]}     Returns an array that looks like this:
  [
    {
      id: 'ssg-ssr',
      title: 'When to Use Static Generation v.s. Server-side Rendering',
      date: '2020-01-01'
    },
    {
      id: 'pre-rendering',
      title: 'Two Forms of Pre-rendering',
      date: '2020-01-02'
    }
  ]
 */
```